File: BHAVRP_Java/BHAVRPv2.0/BHAVRP2.0/src/cujae/inf/ic/om/assignment/classical/ByNotUrgency.java
package cujae.inf.ic.om.assignment.classical;

import java.util.ArrayList;

import cujae.inf.ic.om.problem.input.Customer;

public abstract class ByNotUrgency extends Heuristic {
	
	/**
	 * @param  ArrayList<Customer>
	 * @param  double demanda cubierta del cluster
	 * @param  double capacidad del depósito
	 * @return boolean
	 * Determina si existen clientes que puedan ser asignados al depósito a partir de su demanda
	 */
	protected boolean isFullDepot(ArrayList<Customer> customers, double requestCluster, double capacityDepot){
		boolean isFull = true;

		double currentRequest = capacityDepot - requestCluster;

		if(currentRequest > 0)
		{
			int i = 0;

			while((i < customers.size()) && (isFull))
			{
				if(customers.get(i).getRequestCustomer() <= currentRequest)
					isFull = false;
				else
					i++;
			}
		}	

		return isFull;
		
		// cuando no quedan cliente lo saca como lleno ver si comviene en todos los casos
	}

}

File: BHAVRP_Python/cujae/inf/citi/om/assignment/classical/by_not_urgency.py
from typing import List
from .heuristic import Heuristic
from ...problem.input.customer import Customer

class ByNotUrgency(Heuristic):
    
    def __init__(self):
        super().__init__()
    
    # Método que determina si existen clientes que puedan ser asignados al depósito a partir de su demanda.
    def is_full_depot(self, customers: List[Customer], request_cluster: float, capacity_depot: float) -> bool:
        is_full: bool = True
        current_request: float = capacity_depot - request_cluster
        
        if current_request > 0:
            for customer in customers:
                if customer.get_request_customer() <= current_request:
                    is_full = False
                    break
                
        return is_full